examples/multiomics_mogonet/main.py

```diff
@@ -0,0 +1,107 @@
+"""
+Multi-omics Integration via Graph Convolutional Networks for Biomedical Data Classification.
+
+Reference:
+Wang, T., Shao, W., Huang, Z., Tang, H., Zhang, J., Ding, Z., Huang, K. (2021). MOGONET integrates multi-omics data
+using graph convolutional networks allowing patient classification and biomarker identification. Nature communications.
+https://www.nature.com/articles/s41467-021-23774-w
+"""
+
+import argparse
+import warnings
+
+import pytorch_lightning as pl
+import torch
+from config import get_cfg_defaults
+from model import MogonetModel
+
+import kale.prepdata.tabular_transform as T
+import kale.utils.seed as seed
+from kale.loaddata.multiomics_gnn_dataset import MogonetDataset
+
+warnings.filterwarnings(action="ignore")
+
+
+def arg_parse():
+    """Parsing arguments"""
+    parser = argparse.ArgumentParser(description="MOGONET Training for Multi-omics Data Integration")
+    parser.add_argument("--cfg", required=True, help="path to config file", type=str)
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    args = arg_parse()
+
+    # ---- setup device ----
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    print("\n==> Using device " + device)
+
+    # ---- setup configs ----
+    cfg = get_cfg_defaults()
+    cfg.merge_from_file(args.cfg)
+    cfg.freeze()
+    seed.set_seed(cfg.SOLVER.SEED)
+
+    # ---- setup dataset ----
+    print("\n==> Preparing dataset...")
+    file_names = []
+    for view in range(1, cfg.DATASET.NUM_VIEW + 1):
+        file_names.append(f"{view}_tr.csv")
+        file_names.append(f"{view}_lbl_tr.csv")
+        file_names.append(f"{view}_te.csv")
+        file_names.append(f"{view}_lbl_te.csv")
+
+    multiomics_data = MogonetDataset(
```

Comment:
Call the dataset by a method name is a bit weird.

Reply:
This is also a verbose way to read the dataset. See how neat is Hao's example `train_dataset = PolypharmacyDataset(cfg.DATASET, mode="train")`